core/src/test/scala/integration/kafka/zk/ZkMigrationIntegrationTest.scala

```diff
@@ -950,16 +980,47 @@ class ZkMigrationIntegrationTest {
     dataOpt.map(ProducerIdBlockZNode.parseProducerIdBlockData).get
   }
 
-  def alterTopicConfig(admin: Admin): AlterConfigsResult = {
+  def alterBrokerConfigs(admin: Admin): Unit = {
+    val defaultBrokerResource = new ConfigResource(ConfigResource.Type.BROKER, "")
+    val defaultBrokerConfigs = Seq(
+      new AlterConfigOp(new ConfigEntry(KafkaConfig.LogRetentionTimeMillisProp, "86400000"), AlterConfigOp.OpType.SET),
+    ).asJavaCollection
+    val broker0Resource = new ConfigResource(ConfigResource.Type.BROKER, "0")
+    val broker1Resource = new ConfigResource(ConfigResource.Type.BROKER, "1")
+    val specificBrokerConfigs = Seq(
+      new AlterConfigOp(new ConfigEntry(KafkaConfig.LogRetentionTimeMillisProp, "43200000"), AlterConfigOp.OpType.SET),
+    ).asJavaCollection
+
+    TestUtils.retry(60000) {
+      val result = admin.incrementalAlterConfigs(Map(
+        defaultBrokerResource -> defaultBrokerConfigs,
+        broker0Resource -> specificBrokerConfigs,
+        broker1Resource -> specificBrokerConfigs
+      ).asJava)
+      try {
+        result.all().get(10, TimeUnit.SECONDS)
+      } catch {
+        case t: Throwable => fail("Alter Broker Configs had an error", t)
+      }
+    }
```

Comment:
Do we really need the retry logic here? Could we verify that a KRaft controller is ready before testing this operation to make the outcome more predictable? Or is it expected that we may have to repeat the operation a few times regardless? And if that's the case should we update docs too to let operators know they might have to try a few times?

Reply:
> Could we verify that a KRaft controller is ready before testing this operation to make the outcome more predictable?

I tried this at first, but we don't actually expose this to the client (whether the controller is ZK or KRaft). In fact, we lie to the client and tell it that a random broker is the controller. I could have "pierced the veil" to access the underlying controller classes, but a retry seemed easier.

Reply:
That makes sense. I'm just concerned this retry could hide instability in the operation we're trying to test. I'm thinking we could bypass the client API and retry just on the precondition of the controller being KRaft by e.g. reading the /controller znode, and then expect `incrementalAlterConfigs` to succeed at once. WDYT?

Reply:
I have the same question here.
1. In the test case `testIncrementalAlterConfigsPreMigration`, it should pass at once since KRaft quorum is not ready.
2. In the test `testDualWrite`,  it should pass at once as we have wait for migration to begin.